### PR TITLE
Refactor: remove the use of global variable in test_sandbox

### PR DESCRIPTION
**Pull request by xingyaoww**
**What is the problem that this fixes or functionality that this introduces? Does it fix any open issues?**

Hit some issue with global var in `test_sandbox` in https://github.com/OpenDevin/OpenDevin/pull/2933
---

**Give a summary of what the PR does, explaining any non-trivial design decisions**

- Remove the use of global variable in `test_sandbox`
- Fix an undefined variable

---
**Other references**

### CODE REVIEW

File: tests/unit/test_sandbox.py [hunk 1]
```diff
@@ -36,9 +35,23 @@ def temp_dir(monkeypatch):
 
 def test_env_vars(temp_dir):
     os.environ['SANDBOX_ENV_FOOBAR'] = 'BAZ'
+    ssh_box_config = AppConfig(
```
Comment (tobitege) [inline tests/unit/test_sandbox.py:38]: why not move this into the `create_docker_box_from_app_config` method or even outside/above first test?

**Review by tobitege — CHANGES_REQUESTED** [review]
Lots of duplication, this should be done simpler, imo

**Comment by xingyaoww** [comment]
@tobitege updated - tried to reduce the duplication by making `config` argument optional - if not provided, it will construct a ssh-box by default.

**Review by tobitege — APPROVED** [review]
Looks much better. Not sure if config.py already should (if not already done so), fill the sandbox config with defaults, but for another time. 😀 
Thanks!

**Pull request merged**